Node.js/GettingStarted/azure_batch_client.js

```diff
@@ -0,0 +1,196 @@
+// Initializing node.js modules
+
+var batch = require('azure-batch');
+var https = require('https');
+
+var fs = require('fs');
+var uuid = require('node-uuid');
+
+var accountName = '<azure-batch-account-name>';
+var accountKey = '<account-key-downloaded>';
+var accountUrl = '<account-url>'
+// Setting up Job configuration along with preparation task
+var jobId = "processcsvjob"; 
+
+
+// Create Batch credentials object using account name and account key
+ var credentials = new batch.SharedKeyCredentials(accountName,accountKey);    
+
+ // Create Batch service client
+
+var batch_client = new batch.ServiceClient(credentials,accountUrl);
+
+var now = new Date();
+
+
+var poolid = "processcsv_"+ now.getFullYear() + now.getMonth() + now.getDay() + now.getHours() + now.getSeconds()
+
+console.log("Creating pool with ID" + poolid);
+
+var imgRef = {publisher:"Canonical",offer:"UbuntuServer",sku:"14.04.2-LTS",version:"latest"}
+var vmconfig = {imageReference:imgRef,nodeAgentSKUId:"batch.node.ubuntu 14.04"}
+var numVms = 4;
+var vmSize = "STANDARD_A1";
+
+var clientPoolRequestID = uuid.v4();
+var poolConfig = {id:poolid, displayName:poolid,vmSize:vmSize,virtualMachineConfiguration:vmconfig,targetDedicated:numVms,enableAutoScale:false }
+var poolAddOptionsConfig = {clientRequestId:clientPoolRequestID,returnClientRequestId:true}
+var poolOptions = {poolAddOptions:poolAddOptionsConfig}
+var pool = batch_client.pool.add(poolConfig,poolOptions,function(error,result,request,response){            
+    //        console.log("Pool add response stream" + response.body);
+            if(error != null)
+            {
+                console.log(error.response);
+            }
+            
+        });
+
+console.log("Submitted request to create pool...");
+
+
+var isPoolStateActive = false;
+var poolCreationError = false;
+var maxRetry = 1000;
+var numRetries = 0;
+var jobCreated = false;
+
+while(isPoolStateActive == false && poolCreationError == false && numRetries < maxRetry)
+{
+    
+    var cloudPool = batch_client.pool.get(poolid,function(error,result,request,response){
+        if(error == null)
+        {
+            if(result.state == "active")
+            {
+                  isPoolStateActive = true;
+                  if(jobCreated==true)
+                  {
+                      // do nothing
+                  }
+                  else
+                  {
+                     jobCreated = true;
+                     console.log("Pool created successfully now submitting job");
+                     createJob();
+                  }
+                  //// Create the Job 
+            }
+        }
+        else
+        {
+            if(error.statusCode==404)
+            {
+                //console.log("Pool not found yet returned 404...");    
+                           
+            }
+            else
+            {
+                poolCreationError = true;
+            }
+        }
+        });
```

Comment:
I feel like the indentation has gotten out of sync here - should this be less indented?  Should line 93 be indented by 4 spaces instead of 2?